Framework/DataHandling/src/H5Util.cpp

```diff
@@ -395,6 +397,66 @@ template <typename NumT> std::vector<NumT> readArray1DCoerce(DataSet &dataset) {
   throw DataTypeIException();
 }
 
+/// Test if a group exists in an HDF5 file or parent group.
+bool groupExists(H5::H5Object &h5, const std::string &groupPath) {
+  bool status = true;
+  // Unfortunately, this is actually the approach recommended by the HDF Group.
+  try {
+    h5.openGroup(groupPath);
+  } catch (const H5::Exception &e) {
+    UNUSED_ARG(e);
+    status = false;
+  }
+  return status;
+}
+
+/// Test if an attribute is present on an HDF5 group or dataset and has a specific string value.
+bool keyHasValue(H5::H5Object &h5, const std::string &key, const std::string &value) {
+  bool status = true;
+  try {
+    Attribute attr = h5.openAttribute(key);
+    std::string value_;
+    attr.read(attr.getDataType(), value_);
+    if (value_ != value)
+      status = false;
+  } catch (const H5::Exception &e) {
+    UNUSED_ARG(e);
+    status = false;
+  }
+  return status;
+}
+
+void copyGroup(H5::H5Object &dest, const std::string &destGroupPath, H5::H5Object &src,
+               const std::string &srcGroupPath) {
+  // Source group must exist, and destination group must not exist.
+  if (!groupExists(src, srcGroupPath) || groupExists(dest, destGroupPath))
+    throw std::invalid_argument(std::string("H5Util::copyGroup: source group '") + srcGroupPath + "' must exist and " +
+                                "destination group '" + destGroupPath + "' must not exist.");
+
+  // TODO: check that source file must have at least read access and destination file must have write access.
```

Comment:
I wish I knew offhand where in mantid we already check this.  (At least I hope we do)

Reply:
I don't think it would be checked.  This is a _new_ _utility_ method, and any issue with the permissions would be due to a _usage_ error on the developer's part.  

I left this as a "TODO" because I couldn't find a straightforward way to check this (i.e without trying and failing).  (Unfortunately, a lot of the recommended approaches by the HDF5 group involve trying and failing.)  

What I suggest as a work around would be to include the error codes in the exceptions, as above.  Then at least it would be possible to diagnose what was going on.